src/main/java/Parser.java

```diff
@@ -0,0 +1,132 @@
+        Command currCommand;
+        String taskDescription;
+        // Identify command by first word
+        switch (userInputArr.get(0)) {
+        case "bye":
+            currCommand = new ExitCommand();
+            break;
+        case "list":
+            currCommand = new ListCommand();
+            break;
+        case "todo":
+            // Empty description
+            if (userInputArr.size() == 1) {
+                throw new EmptyDescriptionException("todo");
+            } else {
+                // Splice back the description
+                taskDescription = String.join(" ",
+                        userInputArr.subList(1, userInputArr.size()));
+                currCommand = new AddCommand("todo",
+                        taskDescription, false);
+            }
+            break;
+        case "find":
+            // Empty description
+            if (userInputArr.size() == 1) {
+                throw new EmptyDescriptionException("find");
+            } else {
+                // Splice back the description
+                String findDescription = String.join(" ",
+                        userInputArr.subList(1, userInputArr.size()));
+                currCommand = new FindCommand(findDescription);
+            }
+            break;
+        case "deadline":
+            currCommand = parseSubCommand("deadline", "/by", userInputArr);
+            break;
+        case "event":
+            currCommand = parseSubCommand("event", "/at", userInputArr);
+            break;
+        case "done":
+            // Empty/no list index of task provided
+            if (userInputArr.size() == 1 || userInputArr.size() > 2) {
+                throw new DukeException("Please put the list index of the " +
+                        "completed task after \"done\" and nothing else.");
+            } else {
+                // Check if integer is provided
+                try {
+                    int doneIdx = Integer.parseInt(userInputArr.get(1));
+                    // Check if integer is within range of number of tasks
+                    if (doneIdx > taskList.getNumTasks() || doneIdx < 1) {
+                        throw new DukeException("Integer is not within range of tasks.");
+                    }
+                    currCommand = new DoneCommand(doneIdx);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("Please enter a valid integer after \"Done\".");
+                }
+            }
+            break;
+        case "delete":
+            // Empty/no list index of task provided
+            if (userInputArr.size() == 1 || userInputArr.size() > 2) {
+                throw new DukeException("Please put the list index of the " +
+                        "completed task after \"delete\" and nothing else.");
+            } else {
+                // Check if integer is provided
+                try {
+                    int deleteIdx = Integer.parseInt(userInputArr.get(1));
```

Comment:
Maybe it would reduce nesting if the logic of the commands was refactored into the class files of those commands.